core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -354,11 +353,18 @@ private synchronized Map<PlanNodeId, TaskSource> updateSources(List<TaskSource>
         }
 
         // update maxAcknowledgedSplit
-        maxAcknowledgedSplit = sources.stream()
-                .flatMap(source -> source.getSplits().stream())
-                .mapToLong(ScheduledSplit::getSequenceId)
-                .max()
-                .orElse(maxAcknowledgedSplit);
+        for (TaskSource taskSource : sources) {
+            long maxAcknowledgedSplit = taskSource.getSplits().stream()
+                    .mapToLong(ScheduledSplit::getSequenceId)
+                    .max()
+                    .orElse(Long.MIN_VALUE);
```

Comment:
If taskSource.getSplits().size() == 0 is already checked, .orElse(Long.MIN_VALUE) is not needed. Instead of that, just '.getAsLong()'.